modules/README.md

```diff
@@ -202,18 +194,46 @@ The possible cases for a transaction to revert are:
+
+```solidity
+mapping(bytes32 => ISignatureValidator) validatorForDomain;
+function isValidSignature(bytes32 hash, bytes memory payload) external view returns (bytes4) {
+  (bytes32 domainHash, bytes32 messageHash, bytes memory signature) = abi.decode(payload, (bytes32,bytes32,bytes));
+  require(hash, keccak256(0x19, 0x01, domainHash, messageHash), "Could not validate data");
+  ISignatureValidator validator = validatorForDomain[domainHash];
+  return validator.isValidSignature(msg.sender /* account */, _msgSender(), domainHash, messageHash, signature);
+}
+```
+
+To distinguish signatures that follow this format from "default" signatures the signature is prepended with a special bytes4 identifier. When the dapp requests the signature the wallet can then create the signature in the required format for the correct signature routing.
+
+
 The Layout of the encoded data received by the signature validator is expected to be as follows:
 
 | Start                                                       | End                                                         | Description               |
 |-------------------------------------------------------------|-------------------------------------------------------------|---------------------------|
-| 0x00                                                        | 0x04                                                        | 4 bytes function selector |
-| 0x04                                                        | 0x24                                                        | dataHash                  |
-| 0x24                                                        | 0x44                                                        | domainSeparator           |
-| 0x44                                                        | 0x64                                                        | typeHash                  |
-| 0x64                                                        | 0x84                                                        | encodeData length         |
-| 0x84                                                        | <0x84 + encodeData length>                                  | encodeData                |
-| <0x84 + encodeData length>                                  | <0x84 + encodeData length> + 0x20                           | signature length          |
-| <0x84 + encodeData length> + 0x20                           | <0x84 + encodeData length> + 0x20 + signature length        | signature                 |
+    | 0x00                                                        | 0x04                                                        | 4 bytes selector |
```

Comment:
In the example above we propose `abi.encode`, here we use some custom encoding. Which one is it?

Reply:
removed the table with solidity code showing how encoding should be done